### DISCUSSION TITLE
issue and MR templates API - valid values are not documented

### DISCUSSION THREAD

**Discussion by dazza-codes** [python-gitlab/python-gitlab#2265] [Q&A] [answered]
https://docs.gitlab.com/ee/api/projects.html#templates-for-issues-and-merge-requests
- this is available in the API response for gitlab premium
- it does not document the valid values

https://docs.gitlab.com/ee/user/project/description_templates.html#set-a-default-template-for-merge-requests-and-issues
- this documents two methods to add templates to a project:
    - 1. put files into `.gitlab/**/default.md` for each kind of template
    - 2. edit a UI text box to add template text to it - these UI features are only present in gitlab premium

The question about the gitlab API is what kind of values are allowed, e.g.

```python

mr_template_file = ".gitlab/merge_request_templates/default.md"
mr_template_text = "Some useful MR template text in Markdown format"

if project.merge_commit_template is None:
   
    # refer to option 1 solution because the text is in the file
    project.merge_commit_template = mr_template_file

    # or assign the actual text as if using the UI text box ?
    project.merge_commit_template = mr_template_text

```
 
It seems like the latter might be the API way of replicating the UI text boxes.

**Comment by dazza-codes** [comment, marked as answer]
The `project.merge_commit_template = mr_template_text` works.  The text values also applies to other templates (issues and merge requests).

**Reply by nejch** [reply]
If this has worked for you, that's great - but keep in mind that `merge_commit_template` refers to the Merge commit message template (strings in a UI field, see https://docs.gitlab.com/ee/user/project/merge_requests/commit_templates.html) and has nothing to do with Merge Request Description templates (files in the repository, see https://docs.gitlab.com/ee/user/project/description_templates.html).

In any case it's best to follow upstream API docs for POST/PUT requests if you're unsure about a specific API attribute.
